tensorflow_probability/python/distributions/moyal_test.py

```diff
@@ -0,0 +1,316 @@
+
+    self.assertAllClose(self.evaluate(moyal.stddev()),
+                        stats.moyal.std(loc=loc, scale=scale))
+
+  def testMoyalMode(self):
+    batch_size = 6
+    loc = np.array([[2.0, 4.0, 5.0]] * batch_size, dtype=self.dtype)
+    scale = np.array([1.0], dtype=self.dtype)
+
+    moyal = tfd.Moyal(
+        loc=self.make_tensor(loc),
+        scale=self.make_tensor(scale),
+        validate_args=True)
+
+    self.assertAllClose(self.evaluate(moyal.mode()), self.evaluate(moyal.loc))
+
+  def testMoyalSample(self):
+    loc = self.dtype(4.0)
+    scale = self.dtype(1.0)
+    n = int(3e5)
+
+    moyal = tfd.Moyal(
+        loc=self.make_tensor(loc),
+        scale=self.make_tensor(scale),
+        validate_args=True)
+
+    samples = moyal.sample(n, seed=test_util.test_seed())
+    sample_values = self.evaluate(samples)
+    self.assertEqual((n,), sample_values.shape)
+    self.assertAllClose(
+        stats.moyal.mean(loc=loc, scale=scale),
+        sample_values.mean(), rtol=.01)
+    self.assertAllClose(
+        stats.moyal.var(loc=loc, scale=scale),
+        sample_values.var(), rtol=.01)
+
+  def testMoyalSampleMultidimensionalMean(self):
+    batch_size = 6
+    loc = np.array([[2.0, 4.0, 5.0]] * batch_size, dtype=self.dtype)
+    scale = np.array([1.0, 0.8, 0.5], dtype=self.dtype)
+    n = int(1e5)
+
+    moyal = tfd.Moyal(
+        loc=self.make_tensor(loc),
+        scale=self.make_tensor(scale),
+        validate_args=True)
+
+    samples = moyal.sample(n, seed=test_util.test_seed())
+    sample_values = self.evaluate(samples)
+    self.assertAllClose(
+        stats.moyal.mean(loc=loc, scale=scale),
+        sample_values.mean(axis=0),
+        rtol=.03,
+        atol=0)
+
+  def testMoyalSampleMultidimensionalVar(self):
+    batch_size = 6
+    loc = np.array([[2.0, 4.0, 5.0]] * batch_size, dtype=self.dtype)
+    scale = np.array([1.0, 0.8, 0.5], dtype=self.dtype)
+    n = int(1e5)
+
+    moyal = tfd.Moyal(
+        loc=self.make_tensor(loc),
+        scale=self.make_tensor(scale),
+        validate_args=True)
+
+    samples = moyal.sample(n, seed=test_util.test_seed())
+    sample_values = self.evaluate(samples)
+    self.assertAllClose(
+        stats.moyal.var(loc=loc, scale=scale),
+        sample_values.var(axis=0),
+        rtol=.03,
+        atol=0)
+
+  def testMoyalMoyalKL(self):
+    a_loc = np.arange(-2.0, 3.0, 1.0)
+    a_scale = np.arange(0.5, 2.5, 0.5)
+    b_loc = 2 * np.arange(-2.0, 3.0, 1.0)
+    b_scale = np.arange(0.5, 2.5, 0.5)
+
+    # This reshape is intended to expand the number of test cases.
+    a_loc = a_loc.reshape((len(a_loc), 1, 1, 1))
+    a_scale = a_scale.reshape((1, len(a_scale), 1, 1))
+    b_loc = b_loc.reshape((1, 1, len(b_loc), 1))
+    b_scale = b_scale.reshape((1, 1, 1, len(b_scale)))
+
+    a = tfd.Moyal(loc=a_loc, scale=a_scale, validate_args=True)
+    b = tfd.Moyal(loc=b_loc, scale=b_scale, validate_args=True)
+
+    true_kl = (0.5 * (np.power(2, a_scale/b_scale) *
```

Comment:
I might remove the true_kl computation. The computation itself is a bit complicated and I think the checking sample statistics (i.e. sample KL is consistent with analytic KL) is probably good enough.